private/buf/buflsp/imports.go

```diff
@@ -0,0 +1,183 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file defines wrappers over Buf CLI constructs for searching for imports.
+
+package buflsp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"slices"
+	"strings"
+
+	"github.com/bufbuild/buf/private/bufpkg/bufconfig"
+	"github.com/bufbuild/buf/private/pkg/normalpath"
+	"github.com/bufbuild/buf/private/pkg/storage"
+	"go.lsp.dev/protocol"
+	"go.uber.org/zap"
+)
+
+// configs is the result of findConfigs(). See that function for more information.
+type configs struct {
+	yaml         bufconfig.BufYAMLFile
+	dir          protocol.URI
+	workspace    bufconfig.BufWorkYAMLFile
+	workspaceDir protocol.URI
+}
+
+// findConfigs finds the Buf configuration files in-scope for the given URI.
+func findConfigs(
```

Comment:
Worried that this is duplicating search logic that we have, I want that search logic reused from one place @doriable

Reply:
Yeah, this is something I noted in a list of things to fix/refactor -- this is on me, when I put together `buftarget`, I did not do a good job of making it portable. I should have more aggressively extricated the bucket semantics from `buffetch`, and then that would make the call-site for `buftarget` at the `WorkspaceProvider` layer, and then that would make it so that the logic can be ported here.